base-template.sh

```diff
@@ -484,13 +484,27 @@ run_hook_file() {
 #   1 in case a hook fails, 0 otherwise
 #####################################################
 process_shared_hooks() {
-    SHARED_REPOS_LIST="$1"
-    shift
+    SHARED_HOOKS_TYPE="$1"
+    SHARED_REPOS_LIST="$2"
+    shift 2
 
     update_shared_hooks_if_appropriate "$@"
     execute_shared_hooks "$@" || return 1
 }
 
+#####################################################
+# Check is not a supported git clone url and
+#   is treated as local path.
+# Returns:
+#   none
+#####################################################
+is_local_url() {
+    if echo "$1" | grep -Eq "[a-zA-Z]+://"; then
```

Comment:
Now that I look at this again, I suppose it's still too strict. Perhaps something like this?
```suggestion
    if echo "$1" | grep -Eq "[^/]+://"; then
```

Reply:
Why you mean? :-) Is there such a obscure protocol string?

Reply:
`socks5://` for example, not sure it's something you can actually use with Git, but I have no idea what are all the things you can give to it. 🙂

Reply:
https://git-scm.com/docs/git-clone#_git_urls

Reply:
On that page it says:
> When Git doesn’t know how to handle a certain transport protocol, it attempts to use the remote-<transport> remote helper, if one exists.

My reading of this is that you can have `myF4ncyTr4sp0rt://` URLs and Git will use a helper to handle that.

Reply:
Ah good, dint read it carefully :-). Ok lets do this :-)